graphql-kotlin-schema-generator/src/main/kotlin/com/expedia/graphql/generator/types/ObjectBuilder.kt

```diff
@@ -34,11 +35,10 @@ internal class ObjectBuilder(generator: SchemaGenerator) : TypeBuilder(generator
                 builder.withInterface(GraphQLTypeReference(interfaceType.name))
             } else {
                 kClass.getValidSuperclasses(config.hooks)
-                    .map { objectFromReflection(it.createType(), false) }
+                    .map { graphQLTypeOf(type = it.createType(), inputType = false, annotatedAsID = false) }
                     .forEach {
-                        when (it) {
-                            is GraphQLInterfaceType -> builder.withInterface(it)
```

Comment:
This was the bug. The type would sometimes come back as a GraphQLNonNull if we are now calling `objectFromReflection ` instead of `graphQLTypeOf`. That means this conditional would fail and it would never be added.

Now that we are calling `objectFromReflection`, all the hooks will be called as well

Reply:
i dont think the comment is correct -> when building super classes it would build the interface correctly and add it to the type BUT since it was skipping the `graphQLTypeOf` and instead of doing the direct call to `objectFromReflection` it would also skip all the hooks that are defined in `graphQLTypeOf` method

Reply:
Updated with the new logic